### ISSUE TITLE
Assemblies should include sample name as part of the file name

### ISSUE THREAD

**Issue by apetkau** [phac-nml/irida-galaxy-importer#26]
For assemblies imported into Galaxy, they currently are named `contigs.fasta`.

![image](https://user-images.githubusercontent.com/200517/166970047-24e341d0-9c9a-4c75-852d-9ebcddd084d3.png)

However, in IRIDA the assembly has the sample name included as part of the file name:

![image](https://user-images.githubusercontent.com/200517/166970170-8829ffc8-a30d-4c98-bafb-6c0ba56f1791.png)

This is likely due to this particular line of code:

https://github.com/phac-nml/irida-galaxy-importer/blob/8b85bad43272d523f6b0cf50d9fcce355d3b966e/irida_import/irida_import.py#L182

This is grabbing the `fileName` value instead of the `label` from the REST API. The `label` should have the label which includes the sample name: https://phac-nml.github.io/irida-documentation/developer/rest/#assemblies

This is called from this bit of code:

https://github.com/phac-nml/irida-galaxy-importer/blob/8b85bad43272d523f6b0cf50d9fcce355d3b966e/irida_import/irida_import.py#L108

Since this same function is used for getting sequence files from a sample, a bit of additional logic will likely need to be added to decide if the file is an assembly file.

**Comment by ericenns** [comment]
One thing to note that the files are named `contigs.fasta` because of the Assembly and Annotation pipeline. Should we also consider updating that to prefix the file with the sample name?

**Comment by apetkau** [comment]
Thanks @ericenns . This is actually what the `label` should be doing already. At least I think that's where it is, I can't find the exact code right now. But we did add a feature to prefix output files with the sample name and store this in the database.

It's not the file name saved on the filesystem, but the prefix  (or the full label) should be used when naming files returned to a user.

**Comment by apetkau** [comment]
Here is an example of one place where this is used to rename the analysis output files sent to a user to include the sample name (stored as the label).

https://github.com/phac-nml/irida/blob/d4dfbbdf536bb673df08657f46acc7266ea248e2/src/main/java/ca/corefacility/bioinformatics/irida/ria/utilities/FileUtilities.java#L220-L223

This executes this line to add the prefix for an output file name:

https://github.com/phac-nml/irida/blob/d4dfbbdf536bb673df08657f46acc7266ea248e2/src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/AnalysisOutputFile.java#L146

**Comment by ericenns** [comment]
Yeah I saw that for analysis output files, but for assembly files they actually show up as `contig.fasta` vs uploaded assembly files that actually use the name uploaded i.e. `myuploadedfile.fasta`. I guess the question is do we want to relabel those files? We are not doing this for sequence data (i.e. fastq files).

**Comment by apetkau** [comment]
I may be misunderstanding, but isn't this what we already do?

For genomes that are uploaded, the `getLabel()` returns the filename: https://github.com/phac-nml/irida/blob/development/src/main/java/ca/corefacility/bioinformatics/irida/model/assembly/GenomeAssembly.java#L66-L69

For genomes that are assembled in IRIDA, the `getLabel()`  forwards the request to the `AnalysisOutputFile.getLabel()` which adds the prefix so that `contigs.fasta` will be returned as `sample-contigs.fasta`: https://github.com/phac-nml/irida/blob/development/src/main/java/ca/corefacility/bioinformatics/irida/model/assembly/GenomeAssemblyFromAnalysis.java#L76-L78

**Comment by ericenns** [comment]
My mistake, I was under the wrong impression that they showed up as `contigs.fasta` in IRIDA, but I now see that show up as `SAMPLENAME-contigs.fasta`.

**Comment by apetkau** [comment]
I believe the other issue is with this line of code:

https://github.com/phac-nml/irida-galaxy-importer/blob/8b85bad43272d523f6b0cf50d9fcce355d3b966e/irida_import/irida_import.py#L563

It is not using the sample name as derived from the JSON representation to name the data in Galaxy, but using the physical file name of the filesystem path to name the data in Galaxy.
